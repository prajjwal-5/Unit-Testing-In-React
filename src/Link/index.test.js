import React from 'react';
import renderer from 'react-test-renderer';
import Link from '.';

test('Link changes the class with hovered', () => {
  const component = renderer.create(
    <Link page="http.gmail.com">Facebook</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseEnter();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();


  tree.props.onMouseLeave();

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})