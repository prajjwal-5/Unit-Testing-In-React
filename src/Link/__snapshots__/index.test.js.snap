// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class with hovered 1`] = `
<a
  className="normal"
  href="http.gmail.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link changes the class with hovered 2`] = `
<a
  className="hovered"
  href="http.gmail.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Link changes the class with hovered 3`] = `
<a
  className="normal"
  href="http.gmail.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
